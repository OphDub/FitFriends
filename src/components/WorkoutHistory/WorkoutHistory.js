import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getWorkouts } from '../../actions/actionsIndex';
import './WorkoutHistory.css';

export class WorkoutHistory extends Component {
  async componentWillMount () {
    await this.props.getWorkouts();
  }

  renderedWorkouts = () => {
    const workoutVals = Object.values(this.props.workouts);

    return workoutVals.map((workout, key) => {
      const exerciseVals = Object.values(workout.exercises).map((exercise, key) => {
        return (
          <li className="exercise" key={key}>
            <p className="exercise-reps">{exercise.reps}</p>
            <p className="exercise-name">{exercise.exercise}</p>
          </li>
        );
      });

      return (
        <article className="workout" key={key}>
          <h3 className="workout-name"><span className="workout-title">Workout:</span> {workout.workoutName}</h3>
          <p className="workout-desc">{workout.workoutDesc}</p>
          <ul className="exercises">
            <li className="exercises-header">
              <p className="exercises-reps">Reps:</p>
              <p className="exercises-name">Exercises:</p>
            </li>
            {exerciseVals}
          </ul>
        </article>
      );
    });
  }

  render () {
    return (
      <section className="WorkoutHistory">
        {this.renderedWorkouts()}
      </section>
    );
  }
}

const exercise = {
  reps: PropTypes.string.iwqsRequired,
  exercise: PropTypes.string.isRequired
};

const workout = {
  workoutName: PropTypes.string.isRequired,
  workoutDesc: PropTypes.string.isRequired,
  exercises: PropTypes.arrayOf(PropTypes.shape(exercise)).isRequired
};

WorkoutHistory.propTypes = {
  getWorkouts: PropTypes.func,
  workouts: PropTypes.arrayOf(PropTypes.shape(workout))
};

export const mapStateToProps = (state) => ({
  workouts: state.workouts
});

export const mapDispatchToProps = (dispatch) => ({
  getWorkouts: () => dispatch(getWorkouts())
});

export default connect(mapStateToProps, mapDispatchToProps)(WorkoutHistory);