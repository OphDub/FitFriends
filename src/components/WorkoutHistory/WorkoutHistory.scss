.WorkoutHistory {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: lightgrey;
  min-height: 75%;
  padding: 15px;
  margin: 0 10px 10px 10px;
  border-radius: 10px;
}

.workout {
  padding: 10px;
  width: 95%;
  display: flex;
  flex-direction: column;
  background-color: darkgrey;
  margin: 10px;
  border-radius: 5px;
}

.exercises {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  background-color: white;
  border-radius: 5px;
  padding: 0;
  margin: 10px 0;
  overflow-y: scroll;
}

.exercises-header {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  width: 100%;
  border: 1px solid blue;
}

.exercise {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  width: 100%;
  border: 1px solid magenta;
}

.exercise-name {
  margin: 0px;
  border: 1px solid green;
}