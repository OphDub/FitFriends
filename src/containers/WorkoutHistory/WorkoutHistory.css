.WorkoutHistory {
  display: flex;
  flex-direction: column-reverse;
  align-items: center;
  background-color: lightgrey;
  min-height: 75%;
  padding: 15px;
  margin: 0 10px 10px 10px;
  border-radius: 10px; }

.workout {
  padding: 10px;
  width: 95%;
  display: flex;
  flex-direction: column;
  background-color: darkgrey;
  margin: 10px;
  border-radius: 5px; }

.workout-title {
  color: #ffffff; }

.workout-name {
  margin: 10px 0; }

.exercises {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  background-color: white;
  border-radius: 5px;
  padding: 5px;
  margin: 10px 0;
  overflow-y: scroll; }

.exercises-header {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  width: 100%; }

.exercise {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  width: 100%; }

.exercise-reps {
  flex-grow: 1;
  align-self: center; }
