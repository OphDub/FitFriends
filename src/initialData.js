export const mockTeam = [
  {
    name: 'Felipe',
    workoutTally: 6,
    workoutsPosted: 2,
  },
  {
    name: 'Fion',
    workoutTally: 5,
    workoutsPosted: 3,
  },
  {
    name: 'Morgan',
    workoutTally: 3,
    workoutsPosted: 4,
  },
]

export const mockWorkoutHistory = [
  {
    workoutName: 'The Grind',
    workoutDesc: 'Opposite movement workout with some running at the end!',
    exercises: [
      {reps: '10', exercise: 'Burpees'},
      {reps: '8-10', exercise: 'Overhead Press(Barbell)'},
      {reps: '8-10', exercise: 'Standing Bent Row(Barbell)'},
      {reps: '8-10', exercise: 'Standing Kettlebell Swings'},
    ],
  }
]

export const mockUserProfile = {
  userImage: 'http://www.ixgetfit.com/wp-content/uploads/2017/10/Brooke-Wells-350x350-300x300.jpg',
  userName: 'Brooke'
}

export const mockWorkouts = [
  {
    workoutName: 'local sample workout name 1',
    workoutDesc: 'local sample workout desc',
    exercises: [
      { reps: 10, exercise: 'sample exercise1'},
      { reps: 10, exercise: 'sample exercise2'},
    ]
  },
  {
    workoutName: 'local sample workout name 2',
    workoutDesc: 'local sample workout desc',
    exercises: [
      { reps: 10, exercise: 'sample exercise1'},
      { reps: 10, exercise: 'sample exercise2'},
    ]
  },
]