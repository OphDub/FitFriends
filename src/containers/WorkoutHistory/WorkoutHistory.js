import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getWorkouts, deleteWorkoutFromFirebase } from '../../thunks/thunks';
import './WorkoutHistory.css';

export class WorkoutHistory extends Component {
  async componentWillMount () {
    await this.props.getWorkouts();
  }

  deleteWorkout = (e) => {
    console.log(e.target.parentNode);
  }

  renderedWorkouts = () => {
    const workoutVals = Object.values(this.props.workouts);
    const workoutKeys = Object.keys(this.props.workouts);

    return workoutVals.map((workout, key) => {
      const exerciseVals = Object.values(workout.exercises).map((exercise, key) => {
        return (
          <li className="exercise" key={key}>
            <p className="exercise-reps">{exercise.reps}</p>
            <p className="exercise-name">{exercise.exercise}</p>
          </li>
        );
      });

      return (
        <article className="workout" key={key} id={workoutKeys[key]}>
          <h3 className="workout-name"><span className="workout-title">Workout:</span> {workout.workoutName}</h3>
          <p className="workout-desc">{workout.workoutDesc}</p>
          <ul className="exercises">
            <li className="exercises-header">
              <p className="exercises-reps">Reps:</p>
              <p className="exercises-name">Exercises:</p>
            </li>
            {exerciseVals}
          </ul>
          <button className="workout-delete-btn" onClick={this.deleteWorkout}>Delete Workout</button>
        </article>
      );
    });
  }

  render () {
    return (
      <section className="WorkoutHistory">
        {this.renderedWorkouts()}
      </section>
    );
  }
}

const exercise = PropTypes.shape({
  reps: PropTypes.string,
  exercise: PropTypes.string
});

const workout = PropTypes.shape({
  workoutName: PropTypes.string,
  workoutDesc: PropTypes.string,
  exercises: PropTypes.arrayOf(exercise)
});

WorkoutHistory.propTypes = {
  getWorkouts: PropTypes.func,
  workouts: PropTypes.arrayOf(workout)
};

export const mapStateToProps = (state) => ({
  workouts: state.workouts
});

export const mapDispatchToProps = (dispatch) => ({
  getWorkouts: () => dispatch(getWorkouts())
});

export default connect(mapStateToProps, mapDispatchToProps)(WorkoutHistory);