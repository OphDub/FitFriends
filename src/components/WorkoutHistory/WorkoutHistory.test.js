import React from 'react';
import { shallow } from 'enzyme';
import { WorkoutHistory, mapStateToProps, mapDispatchToProps } from './WorkoutHistory';
import { mockWorkoutHistory } from '../../initialData';

describe('WORKOUT HISTORY', () => {
  let renderedComponent;
  let mockFn;

  beforeEach(() => {
    mockFn = jest.fn();
    renderedComponent = shallow(<WorkoutHistory getWorkouts={mockFn} workouts={mockWorkoutHistory}/>);
  });

  it('should match snapshot', () => {
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should map the store correctly', () => {
    const mockStore = {
      workouts: mockWorkoutHistory
    };

    const mapped = mapStateToProps(mockStore);

    expect(mapped.workouts).toEqual(mockStore.workouts);
  });

  it('should call dispatch when using a function from mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);

    mapped.getWorkouts();

    expect(mockDispatch).toHaveBeenCalled();
  });
});