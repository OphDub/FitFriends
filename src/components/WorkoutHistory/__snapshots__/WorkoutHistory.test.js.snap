// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WORKOUT HISTORY should match snapshot 1`] = `undefined`;
